### ISSUE TITLE
gitlab_project_variable overwrite GitLab variables with the same name and different scopes

### ISSUE THREAD

**Issue by caiconkhicon** [gitlabhq/terraform-provider-gitlab#390]
### Terraform Version
0.12.29

### Affected Resource(s)
- gitlab_project_variable

### Terraform Configuration Files
I have a Terraform module which is shared between environments
```hcl
resource "gitlab_project_variable" "foo" {
  project         = "80"
  key               = "FOO"
  value            = "bar"
  protected     = false
  masked        = true
  environment_scope = var.cluster_environment
}
```

### Expected Behavior
For each environment, a GitLab variable with an identical name is created, but different environment scope. 

### Actual Behavior
I apply on `dev` first, then `test`, then this is showed when applying to `test`:
```hcl
   # module.abc.gitlab_project_variable.foo will be updated in-place
   ~ resource "gitlab_project_variable" "foo" {
       ~ environment_scope = "dev" -> "test"
         id                = "80:FOO"
         key               = "FOO"
         masked            = true
         project           = "80"
         protected         = false
         value             = "bar"
         variable_type     = "env_var"
     }
```

**Comment by jeremad** [comment]
https://github.com/gitlabhq/terraform-provider-gitlab/blob/master/gitlab/resource_gitlab_project_variable.go#L88

The issue seems that the unique ID of the terraform resource is based on the project and the key. We should put the scope in there as well, and use the filter parameter of the GitLab API:
https://docs.gitlab.com/ee/api/project_level_variables.html#show-variable-details

**Comment by peric** [comment]
I can confirm I have the same issue 😞 

What @jeremad suggested seems legit to me. Is anyone already working on this?

**Comment by Mayeu** [comment]
There is a lot of discussion about this in #213. There was some fix attempt but none to match the last fix deployed in 13.2 as far as I know.

**Comment by peric** [comment]
I also did some attempt on this, but then I realised:

* Feature `ci_variables_api_filter_environment_scope` needs to be enabled in Gitlab. In my case, this affects many other projects and changes
* Changes mentioned above should be also done in [go-gitlab](https://github.com/xanzy/go-gitlab/blob/master/project_variables.go)

I have prototype of this on my local machine and hopefully I'll manage to test it in the next days. But I'm also not sure how to do it without BC breaks.
